### PR TITLE
[GarbageCollector] Let load and density e2e tests use GC if it's on

### PR DESCRIPTION
**Pull request by caesarxuchao**
I've run the 100 and 500 nodes tests and they both pass.

The test-infra half of the PR is https://github.com/kubernetes/test-infra/pull/369

cc @lavalamp

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/30401)

### CODE REVIEW

**Comment by caesarxuchao** [comment]
@k8s-bot e2e test this, issue #IGNORE

File: test/e2e/density.go [hunk 1]
```diff
@@ -319,9 +319,15 @@ func cleanupDensityTest(dtc DensityTestConfig) {
...
-			By("Cleaning up the replication controller")
-			err := framework.DeleteRC(dtc.Client, dtc.Namespace, rcName)
-			framework.ExpectNoError(err)
+			if framework.TestContext.GarbageCollectorEnabled {
+				By("Cleaning up only the replication controller, garbage collector will clean up the pods")
+				err := framework.DeleteOnlyRC(dtc.Client, dtc.Namespace, rcName)
+				framework.ExpectNoError(err)
+			} else {
+				By("Cleaning up the replication controller and pods")
+				err := framework.DeleteRC(dtc.Client, dtc.Namespace, rcName)
```
Comment (gmarek) [inline test/e2e/density.go]: I think we should rename it to DeleteRCAndPods

File: test/e2e/framework/util.go [hunk 2]
```diff
@@ -3127,7 +3127,41 @@ func DeleteRC(c *client.Client, ns, name string) error {
...
+	if err != nil {
+		return fmt.Errorf("error while deleting RC %s: %v", name, err)
+	}
+	terminatePodTime := time.Now().Sub(startTime) - deleteRCTime
+	Logf("Terminating RC %s pods took: %v", name, terminatePodTime)
+	return nil
+}
+
+// DeleteOnlyRC deletes only the Replication Controller and waits for GC to delete the pods.
+func DeleteOnlyRC(c *client.Client, ns, name string) error {
```
Comment (gmarek) [inline test/e2e/framework/util.go]: DeleteRCOnly

**Comment by gmarek** [comment]
2 naming nits. LGTM otherwise. GKE test endpoint is down, which causes smoke tests to fail.

File: test/e2e/framework/util.go (hunk 2 above)
Comment (caesarxuchao) [inline test/e2e/framework/util.go]: I renamed it to `DeleteRCAndWaitForGC` to be more precise.

**Comment by caesarxuchao** [comment]
@k8s-bot e2e test this.

**Comment by caesarxuchao** [comment]
@k8s-bot e2e test this, issue #IGNORE.

**Comment by caesarxuchao** [comment]
@gmarek I fixed the function names. I'll apply the label myself since the changes are trivial.

**Comment by caesarxuchao** [comment]
@k8s-bot unit test this issue, #30377

**Comment by caesarxuchao** [comment]
@k8s-bot e2e test this, issue #30075

**Pull request merged**